src/main/java/net/snowflake/client/core/bind/BindUploader.java

```diff
@@ -189,13 +196,126 @@ public static synchronized BindUploader newInstance(SFSession session, String st
    * @param bindValues the bind map to upload
    * @throws BindException if the bind map could not be serialized or upload fails
    */
-  public void upload(Map<String, ParameterBindingDTO> bindValues) throws BindException {
+  @Deprecated
+  public void uploadDeprecated(Map<String, ParameterBindingDTO> bindValues) throws BindException {
     if (!closed) {
       serializeBinds(bindValues);
       putBinds();
     }
   }
 
+  /**
+   * Upload bind parameters via streaming. This replaces previous function upload function where
+   * binds were written to a file which was then uploaded with a PUT statement.
+   *
+   * @param bindValues the bind map to upload
+   * @throws BindException if the bind map could not be serialized or upload fails
+   * @throws SQLException if the upload fails
+   */
+  public void upload(Map<String, ParameterBindingDTO> bindValues)
```

Comment:
New upload function fills byte array until buffer size is reached and then triggers uploadStreamInternal function. It iterates until parameter list has all been uploaded.

Reply:
Should we also take a look at https://github.com/snowflakedb/snowflake-jdbc/blob/0c60d25b9e0fbd0b8b2dc939428fc2cdc34a3e49/src/main/java/net/snowflake/client/core/SFStatement.java#L384? Like, if users are not able to write to local disks, I guess they can't create temporary dir as well. Then they will always fail to get BindUploader through `Path bindDir = Files.createTempDirectory(PREFIX);` in BindUploader.newInstance?

Reply:
Great catch!! Yes, this would cause error. I've taken out the creation of a temporary directory. No bindDir is initialized now.